iceoryx_posh/CMakeLists.txt

```diff
@@ -203,6 +203,7 @@ add_library(iceoryx_posh_roudi
     source/roudi/port_manager.cpp
     source/roudi/port_pool.cpp
     source/roudi/roudi.cpp
+    source/roudi/roudi_config.cpp
```

Comment:
What which target was experiencing the linker error ? This file is included in the target `iceoryx_posh_config` so linking to that should resolve it ...

Reply:
Ah, it was in larry, did you try linking it to `iceoryx_posh_config`?

Reply:
@ithier Oh you are right!! ... when we did introduce that target?!

Reply:
At the same time as moving DDS config into posh ... The reason was to isolate the dynamic configuration classes because it that has a lower safety rating to the core posh library (e.g. because of toml).
When factoring out the toml stuff, all of the config classes that are used in conjunction with it were also factored out to avoid circular dependencies.

@mossmaurice FYI